hydra-access-controls/app/models/concerns/hydra/access_controls/permissions.rb

```diff
@@ -437,6 +439,18 @@ def person_agent?(agent)
         raise 'no agent' unless agent.present?
         agent.first.rdf_subject.to_s.start_with?(PERSON_AGENT_URL_PREFIX)
       end
+
+      # Removes any permissions if both a delete and an update are found for the same id
+      # Fixes https://github.com/projecthydra/sufia/issues/2821
+      def remove_concurrent_deletes(collection)
+        collection.each do |permission|
+          next unless has_destroy_flag?(permission)
+          delete_id = permission.fetch(:id, nil)
+          if collection.map { |c| c if c.fetch(:id, nil) == delete_id }.count > 1
```

Comment:
I've added an additional test and delete_if to remove deletes that don't have ids. That was causing unexpected behavior. Regarding `any?`, that doesn't work because the array needs to be larger that 1, indicating that there are two operations for the same id. If one of them is a delete, then they both need to be removed.